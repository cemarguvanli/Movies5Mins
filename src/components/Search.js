import React, { useState, useCallback } from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import { CloseIcon } from './Icons';

import useDebounce from '../hooks/useDebounce';

const Search = ({
  value, onChange, debounceTime
}) => {
  const [searchInputFocussed, setSearchInputFocussed] = useState(false);
  const [searchValue, setSearchValue] = useState(value || '');

  const focusTextInput = useCallback(() => setSearchInputFocussed(true), []);
  const blurTextInput = useCallback(() => setSearchInputFocussed(false), []);
  const handleClearSearch = useCallback(() => setSearchValue(''), []);

  useDebounce(searchValue, (val) => onChange(val), debounceTime);

  return (
    <View style={styles.searchInputWrapper}>
      <TextInput
        autoCorrect={false}
        autoCompleteType='off'
        placeholder="Search movies..."
        value={searchValue}
        onChangeText={(val) => setSearchValue(val)}
        onFocus={focusTextInput}
        onBlur={blurTextInput}
        style={styles.searchInputStyle}
        placeholderTextColor="#8E8E93"
      />
      {(searchInputFocussed && searchValue.length > 0) && (
        <TouchableOpacity style={styles.closeIconWrapper} onPress={handleClearSearch}>
          <CloseIcon />
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  searchInputStyle: {
    fontSize: 16,
    fontWeight: '400',
    height: 50,
    letterSpacing: 0.5,
    paddingHorizontal: 24,
    paddingVertical: 10,
    paddingRight: 36,
    borderRadius: 12,
    width: '100%',
    backgroundColor: 'rgba(142,142,147,0.12)'
  },
  searchInputWrapper: {
    paddingLeft: 16,
    paddingRight: 16,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    zIndex: 10,
    marginBottom: 32,
  },
  closeIconWrapper: {
    position: 'absolute',
    right: 28,
    top: 18
  },
});

export default Search;