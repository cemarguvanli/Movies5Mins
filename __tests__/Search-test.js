
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Search from '../src/components/Search';

describe('Manually mocking functions', () => {
  it('renders correctly across screens', () => {
    const tree = renderer.create(<Search debounceTime={500} onChange={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});