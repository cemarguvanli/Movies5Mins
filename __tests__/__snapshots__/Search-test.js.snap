// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manually mocking functions renders correctly across screens 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 32,
      "paddingLeft": 16,
      "paddingRight": 16,
      "zIndex": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="off"
    autoCorrect={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Search movies..."
    placeholderTextColor="#8E8E93"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "rgba(142,142,147,0.12)",
        "borderRadius": 12,
        "fontSize": 16,
        "fontWeight": "400",
        "height": 50,
        "letterSpacing": 0.5,
        "paddingHorizontal": 24,
        "paddingRight": 36,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
